### DISCUSSION TITLE
Dichotomy: User has to choose between potential DOS attack or potential security vulnerability

### DISCUSSION THREAD

**Discussion by ghost** [anonaddy/anonaddy#385] [General]
An imaginary AnonAddy user likes to create unique aliases for each website that they are registered to. This user also likes to be able to remember their aliases, and so they usually create them in the following format for each of the websites they use: `website-domain`@`anonaddy-username`.anonaddy.com, or something similar. 

Let's say this user had one of their email aliases leaked in a data breach. In fact, imagine that Twitter got hacked and multiple account emails got leaked to the public, including our user's Twitter account email: twitter.com@`anonaddy-username`.anonaddy.com. At first this may not sound like a huge issue, but it just might be, and I'll explain why.

You see, if our user doesn't use a custom domain and instead use one of AnonAddy's (e.g., anonaddy.com), they are unable to disable catch-all email for `anonaddy-username`.anonaddy.com. A bad actor could see our user's email in the leaked database (in this case, twitter.com@`anonaddy-user`.anonaddy.com) and send mass emails to `random`@`anonaddy-user`.anonaddy.com until all of the user's monthly bandwidth is used. Even if our user disables the newly created alias, the bad actor can continue to create new ones on their behalf indefinitely and constantly impede our user from receiving new emails. This opens up the possibility for an AnonAddy user to suffer some kind of "DOS attack" since they are unable to disable catch-all and all non-existing aliases will be created for them automatically, using up all their monthly bandwidth.

However, even if our user could disable catch-all for their username-prefixed subdomain in AnonAddy, they could still face some serious problems if their alias were to get leaked. It is true that if catch-all is disabled a bad actor would have a harder time pullage then that means that user in fact has that alias, which would indicate that they have an account on that website. That's a potential privacy issue, especially if that website allows you to find a user by their email address.

Based on the hypothetical scenario above, I suggest that:
ing off a DOS attack, since the user can simply deactivate the troublesome alias, however with catch-all disabled it's easier to determine what websites our user is registered to: all the bad actor would have to do is send mass emails to `_____`@`anonaddy-username`.anonaddy.com, replacing `_____` with well-known websites such as facebook.com, youtube.com etc, and if the email does not bounce with an error message then that means that user in fact has that alias, which would indicate that they have an account on that website. That's a potential privacy issue, especially if that website allows you to find a user by their email address.

Based on the hypothetical scenario above, I suggest the following steps:

1. Any email sent to an AnonAddy alias should at least "pretend" to go through, meaning that it will never bounce with an error message even if catch-all is disabled, so as to avoid brute-force attacks. 
2. Catch-all should be disabled by default in AnonAddy's tier II (Lite) plan, both for AnonAddy domains and custom domains.
3. If catch-all is disabled, then the email simply doesn't get to our user, even though to the attacker it will always look like it went through, and that the email address exists. That way we can avoid bad actors using up all our bandwidth and trying to discover which websites we use by doing a brute-force attack on the alias custom local part. The user could then keep track of the "contaminated" aliases and decide whether to promote them to "usable" aliases or not.
4. If catch-all is enabled and the user receives an email to an address that is not a registered alias, the email will go through, but the alias will be listed as "unregistered" so that they can know it's not one of their "official" aliases, which could also prevent social engineering attacks. They can later promote the alias to a "registered" state.

**Comment by vertigo220** [comment]
I didn't read all of this (sorry) and I'm not saying reasonable steps shouldn't be taken to prevent it, but I will say that based on using a similar service for probably close to a decade now, I've found this concern to be unnecessary. That's not to say it can't happen, but a major reason for using services like this is to protect against email addresses being sold/shared/stolen, which I've also only experienced a small handful of times in the time I've been using aliases, something I find interesting. If an (aliased) address is sold/etc, that could very well lead to the same scenario, yet it never has for me. The simple fact is, nobody is going to take the time to manually go through a list dozens or hundreds of pages long of email addresses and look for ways to manipulate them like this. Most people don't even bother to use aliases, so it would only be a small percentage in any breached list that this would apply to, and it's not worth the time it would take to do it. Even if they were to just do a search, e.g. for "twitter" in your example, to find such email addresses (which is highly unlikely because even that is just not going to be worth the time), they would likely expect anyone using an alias service is also a) going to be taking other protective measures, and b) not going to be buying what they're selling or not likely to open malicious attachments or whatever the case may be. It's a very logical concern, and one I've had, it's just not a realistic one. So again, reasonable steps to protect against this type of abuse should be taken, but it's not something that should be a priority and it's not something worth worrying about. And I'm sure (or at least would hope) that in the extremely off-chance it were to happen, the admin would work with you on it and not just leave you with an unusable account.
